core/src/main/java/com/linecorp/armeria/server/ExceptionReportingServerErrorHandler.java

```diff
@@ -0,0 +1,152 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.util.TextFormatter;
+import com.linecorp.armeria.server.logging.LoggingService;
+
+import io.micrometer.core.instrument.Tags;
+
+/**
+ * A {@link ServerErrorHandler} that wraps another {@link ServerErrorHandler}
+ * to periodically report the exceptions that were not logged by decorators such as
+ * {@link LoggingService}.
+ */
+final class ExceptionReportingServerErrorHandler implements ServerErrorHandler, ServerListener {
+
+    private static final Logger logger = LoggerFactory.getLogger(ExceptionReportingServerErrorHandler.class);
+
+    @Nullable
+    private static ExceptionReportingServerErrorHandler exceptionReportingServerErrorHandler;
+    private final ServerErrorHandler delegate;
+    private final Duration duration;
+    private final LongAdder counter;
+    private long lastExceptionsCount;
+    @Nullable
+    private Throwable thrownException;
+    @Nullable
+    private ScheduledFuture<?> reportingTaskFuture;
+
+    private ExceptionReportingServerErrorHandler(ServerErrorHandler serverErrorHandler, Duration duration) {
+        assert !duration.isNegative() && !duration.isZero();
+        delegate = serverErrorHandler;
+        this.duration = duration;
+        counter = new LongAdder();
+    }
+
+    static ExceptionReportingServerErrorHandler of(ServerErrorHandler serverErrorHandler, Duration duration) {
+        if (exceptionReportingServerErrorHandler != null) {
+            return exceptionReportingServerErrorHandler;
+        }
+
+        exceptionReportingServerErrorHandler = new ExceptionReportingServerErrorHandler(serverErrorHandler,
+                                                                                        duration);
+        return exceptionReportingServerErrorHandler;
+    }
+
+    /**
+     * This method increments the {@link #counter} and stores unhandled exception that occurs. If unhandled
+     * exceptions are not previously scheduled for reporting, then they are scheduled to be reported.
+     */
+    @Nullable
+    @Override
+    public HttpResponse onServiceException(ServiceRequestContext ctx, Throwable cause) {
+        final boolean isExpectedException =
+                (cause instanceof HttpStatusException || cause instanceof HttpResponseException) &&
+                cause.getCause() != null;
+
+        if (ctx.shouldReportUnhandledExceptions() && !isExpectedException) {
+            if (reportingTaskFuture == null) {
+                reportingTaskFuture = ctx.eventLoop().scheduleAtFixedRate(
+                        this::reportException, duration.toMillis(), duration.toMillis(), TimeUnit.MILLISECONDS);
```

Comment:
As you might have noticed, different servers can have different event loops. You can't use one reporting task for multiple server instances (Imagine the situation where a user starts two different servers with two different event loop groups.) Therefore, you can't use a singleton. Each server must have its own reporting handler.

Reply:
Oh, missed that point. So can we conclude like below? 
- Use server's lifecycle methods instead of using singleton(as multiple servers can be started in single jvm) 
- Use Micrometer counter for metric collection 
- Use LongAdder for reporting unhandled exceptions for each server